verification/policy.go

```diff
@@ -0,0 +1,130 @@
+/*
+Package verification provides the utilities for handling verification related logic
+like Trust Stores and Trust Policies. Few utilities include loading, parsing, and validating
+trust policies and trust stores.
+*/
+package verification
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+// PolicyDocument represents a trust_policy.json document
+type PolicyDocument struct {
+	// Version of the policy document
+	Version string `json:"version"`
+	// TrustPolicies include each policy statement
+	TrustPolicies []TrustPolicy `json:"trustPolicies"`
+}
+
+// TrustPolicy represents a policy statement in the policy document
+type TrustPolicy struct {
+	// Name of the policy statement
+	Name string `json:"name"`
+	// RegistryScopes that this policy statement affects
+	RegistryScopes []string `json:"registryScopes"`
+	// SignatureVerification setting for this policy statement
+	SignatureVerification string `json:"signatureVerification"`
+	// TrustStore this policy stament uses
+	TrustStore string `json:"trustStore,omitempty"`
+	// TrustedIdentities this policy statement pins
+	TrustedIdentities []string `json:"trustedIdentities,omitempty"`
+}
+
+func isPresent(val string, values []string) bool {
+	for _, v := range values {
+		if v == val {
+			return true
+		}
+	}
+	return false
+}
+
+// ValidatePolicyDocument validates a policy document according to it's version's rule set.
+// if any rule is violated, returns an error
+func ValidatePolicyDocument(policyDoc *PolicyDocument) error {
+	// Constants
+	wildcard := "*"
+	supportedPolicyVersions := []string{"1.0"}
+	supportedVerificationPresets := []string{"strict", "permissive", "audit", "skip"}
+	supportedTrustStorePrefixes := []string{"ca", "signingservice"}
+
+	// Validate Version
+	if !isPresent(policyDoc.Version, supportedPolicyVersions) {
+		return fmt.Errorf("version %q is not supported", policyDoc.Version)
+	}
+
+	// Validate the policy according to 1.0 rules
+	if len(policyDoc.TrustPolicies) == 0 {
+		return errors.New("trust policy document can not have zero policy statements")
+	}
+	policyStatementNameCount := make(map[string]int)
+	registryScopeCount := make(map[string]int)
+	for _, statement := range policyDoc.TrustPolicies {
+
+		// Verify statement name is valid
+		if statement.Name == "" {
+			return errors.New("policy statement is missing a name")
+		}
+		policyStatementNameCount[statement.Name]++
+
+		// Verify registry scopes are valid
+		if len(statement.RegistryScopes) == 0 {
+			return fmt.Errorf("policy statement %q has zero registry scopes", statement.Name)
+		}
+		if len(statement.RegistryScopes) > 1 && isPresent(wildcard, statement.RegistryScopes) {
+			return errors.New("wildcard registry scope can not be shared with other registries")
```

Comment:
```suggestion
                        return fmt.Errorf("trust policy statement %q uses wildcard registry scope '*', a wildcard scope cannot be used in conjunction with other scope values, it must be the only single value for registry scope", statement.Name)
			return errors.New("wildcard registry scope can not be shared with other registries")
```

Reply:
Registry scope validation rules could be moved to its own method. Missing the validation on allowed values, either a `*` or fully qualified registry URL without the scheme/protocol. E.g. `registry.acme-rockets.io/software/net-monitor`

Reply:
Will add URL validation. Regarding the error message, will use `trust policy statement %q uses wildcard registry scope '*', a wildcard scope cannot be used in conjunction with other scope values", statement.Name`